src/Grid/EventButton.vala

```diff
@@ -55,17 +57,26 @@ public class Maya.View.EventButton : Gtk.Revealer {
                 Gtk.Menu menu = new Gtk.Menu ();
                 menu.attach_to_widget (this, null);
 
-                var edit_item = new Gtk.MenuItem.with_label (_("Edit…"));
-                var remove_item = new Gtk.MenuItem.with_label (_("Remove"));
+                bool sensitive = src.writable == true && Model.CalendarModel.get_default ().calclient_is_readonly (src) == false;
 
-                if (src.writable != true && Model.CalendarModel.get_default ().calclient_is_readonly (src) != false) {
-                    edit_item.sensitive = false;
-                    remove_item.sensitive = false;
+                Gtk.MenuItem edit_item = new Gtk.MenuItem.with_label (_("Edit…"));
+                edit_item.activate.connect (() => { edition_request (); });
+                edit_item.sensitive = sensitive;
+                menu.append (edit_item);
+
+                Gtk.MenuItem remove_item;
+                if (comp.has_recurrences ()) {
+                    remove_item = new Gtk.MenuItem.with_label (_("Remove series"));
```

Comment:
I'm not sure these are super clear labels. Maybe it would be more clear to have "Remove This Event" and "Remove All Future Events"?

Either way, menu items should be Title Case like buttons :)

Reply:
The problem is we refer to events throughout the app as containing all instances, so we have "Remove Event" throughout removing the entire series, I could accept changing "Remove Series" to "Remove Event" but that seems a bit confusing... "Remove All Future Events" I am very opposed to as that would in my opinion just remove future events, which is a very different behaviour to what this is doing.

Reply:
Can you clarify which each of the buttons do? To me, the terms “series” and “occurrence” aren’t clear. I don’t know what those mean or what will happen when I select one of these items

Reply:
Events can repeat so for example an event can repeat every tuesday and thursday, but they still are the same event just on multiple dates (or datetimes), currently when you remove an event we delete all of the dates, in this pull I am using Series to refer to the whole set of dates referred to by an event which is defined by the recurrence rule (and some other properties). Whereas the Occurence term I am using to refer to on specific date that an event occurs on, so just one of the Tuesdays because you can't do the meeting that day (in reality the meeting would probably be moved which is why we should support RDATES as well as EXDATES) so an exception is added to say that the event will not occur on that date.

Reply:
For a comparison Timepage has "Delete this event" in place of the "Remove Occurrence" and "Delete all occurrences" in place of "Remove Series" however we may need to redesign existing parts of the interface to treat an event as being just one date of an event... iOS seems to similarly treat each date as an event, but then only let you modify future occurrences (and this behaviour seems quite unpredictable even with standard complying events.) Blackberry uses Series and Occurrence like this. Gnome Calendar (at least the version in Loki) only seems to delete Occurrences individually as "Delete Event", but then it also doesn't seem to properly support repeats and they have to be created externally...

Reply:
If we have a precedent in another part of the UI, we should probably stick to using the same labels so we stay consistent with ourselves

Reply:
Another related thing is we may want to stick to either 'remove' or 'delete' if trying to be consistent, but that is another problem.